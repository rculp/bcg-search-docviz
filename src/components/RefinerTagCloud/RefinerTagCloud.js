import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v1';

import { Header } from 'semantic-ui-react';

import './RefinerTagCloud.scss';

const RefinerTagCloud = ({ box }) => (
  <div className="refiner-tag-cloud">
    <Header as="h2">{box.display}</Header>
    <ul>
      { box.items.map(item => <li key={uuid()}>{item.Name}</li>) }
    </ul>
  </div>
);

RefinerTagCloud.propTypes = {
  box: PropTypes.shape({
    display: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape(
      { Name: PropTypes.string }
    ))
  }).isRequired
};

export default RefinerTagCloud;
