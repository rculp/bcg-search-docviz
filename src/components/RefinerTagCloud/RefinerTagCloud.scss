.refiner-tag-cloud {
  z-index: 0;
}
