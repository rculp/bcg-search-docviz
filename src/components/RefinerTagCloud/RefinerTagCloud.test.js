import React from 'react';
import { shallow } from 'enzyme';
import uuid from 'uuid/v1';
jest.mock('uuid/v1', () => (jest.fn(() => 1)));

import RefinerTagCloud from './RefinerTagCloud';

describe('Refiner Tag Cloud', () => {
  const props = {
    box: {
      display: 'displayName',
      items: [
        { Name: 'name1' },
        { Name: 'name2' },
        { Name: 'name3' }
      ]
    }
  };

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <RefinerTagCloud {...props} />
    );
    expect(component).toMatchSnapshot();
  });
});