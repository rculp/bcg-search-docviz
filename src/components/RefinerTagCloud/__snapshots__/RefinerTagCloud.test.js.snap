// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Refiner Tag Cloud renders and matches our snapshot 1`] = `
<div
  className="refiner-tag-cloud"
>
  <Header
    as="h2"
  >
    displayName
  </Header>
  <ul>
    <li
      key="1c4e0e80-7322-11e8-9dc6-b502906aa596"
    >
      name1
    </li>
    <li
      key="1c4e3590-7322-11e8-9dc6-b502906aa596"
    >
      name2
    </li>
    <li
      key="1c4e3591-7322-11e8-9dc6-b502906aa596"
    >
      name3
    </li>
  </ul>
</div>
`;
